### PR TITLE
Update C API pipeline to use CentOS 6

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Update C API pipeline to use CentOS 6.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
-      displayName: 'Build and Test Linux on Docker'
+    - task: CmdLine@2
+      inputs:
+        script: |
+          docker build -t onnxruntime-centos6 --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=3.6 -f Dockerfile.centos6 .
+        workingDirectory: $(Build.SourcesDirectory)/tools/ci_build/github/linux/docker
+    - task: CmdLine@2
+      inputs:
+        script: |
+          docker run --rm --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build -e NIGHTLY_BUILD onnxruntime-centos6 /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_shared_lib --use_openmp --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest --use_automl
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: Do we want to set -e NIGHTLY_BUILD?

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -89,7 +89,9 @@ set(protobuf_BUILD_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
 #nsync tests failed on Mac Build
 set(NSYNC_ENABLE_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
 set(ONNX_ML 1)
-
+if(NOT onnxruntime_ENABLE_PYTHON)
+  set(onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS OFF)
```
Comment (pranavsharma) [inline cmake/CMakeLists.txt]: Isn't this OFF by default at line 84?

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
-      displayName: 'Build and Test Linux on Docker'
+    - task: CmdLine@2
+      inputs:
+        script: |
+          docker build -t onnxruntime-centos6-gpu --build-arg BUILD_USER=onnxruntimedev --build-arg BUILD_UID=$(id -u) --build-arg PYTHON_VERSION=$(python.version) -f Dockerfile.centos6_gpu .
+        workingDirectory: $(Build.SourcesDirectory)/tools/ci_build/github/linux/docker
+    - task: CmdLine@2
+      inputs:
+        script: |
+          docker run --gpus all -e NVIDIA_VISIBLE_DEVICES=all --rm --volume $(Build.SourcesDirectory):/onnxruntime_src --volume $(Build.BinariesDirectory):/build -e NIGHTLY_BUILD onnxruntime-centos6-gpu /usr/bin/python3.6 /onnxruntime_src/tools/ci_build/build.py --build_dir /build --config Release --skip_submodule_sync  --parallel --build_shared_lib --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest --use_automl --use_cuda --cuda_version=10.0 --cuda_home=/usr/local/cuda-10.0  --cudnn_home=/usr/local/cuda-10.0
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: Same question about NIGHTLY_BUILD.

File: cmake/CMakeLists.txt (hunk 2 above)
Comment (snnn) [reply cmake/CMakeLists.txt]: build.py may set it to ON

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: This is passing the env to build.py. If we want to generate a nightly build with pyop functionality, then we need this.

**Pull request merged**